tests/RecordStream/Operation/parsedate.t

```diff
@@ -0,0 +1,139 @@
+use strict;
+use warnings;
+
+use Test::More;
+use App::RecordStream::Test::OperationHelper;
+use App::RecordStream::Operation::parsedate;
+
+BEGIN {
+  # Normalize localtime for testing
+  $ENV{TZ} = 'US/Pacific';
+}
+
+# These tests aim to exercise the interplay of recs-provided options to
+# parsedate to ensure they're working correctly, not test Time::ParseDate's
+# functionality which is proven elsewhere.
+
+note 'Timezones';
+{
+  my @args = qw[ -k when --format %T ];
+
+  App::RecordStream::Test::OperationHelper->do_match(
+    'parsedate',
+    [@args, qw[ --localtime ]],
+    '{"when":"Feb 23 21:51:47 2016"}',
+    '{"when":"21:51:47"}',
+    "Feb 23 21:51:47 2016 (assuming \$ENV{TZ} = $ENV{TZ}) is 21:51:47 PST",
+  );
+
+  App::RecordStream::Test::OperationHelper->do_match(
+    'parsedate',
+    [@args, qw[ --gmtime ]],
+    '{"when":"Feb 23 21:51:47 2016"}',
+    '{"when":"05:51:47"}',
+    "Feb 23 21:51:47 2016 (assuming \$ENV{TZ} = $ENV{TZ}) is 05:51:47 UTC",
+  );
+
+  App::RecordStream::Test::OperationHelper->do_match(
+    'parsedate',
+    [@args, qw[ --tz UTC --localtime ]],
+    '{"when":"Feb 23 21:51:47 2016"}',
+    '{"when":"13:51:47"}',
+    "Feb 23 21:51:47 2016 (with --tz UTC) is 13:51:47 PST",
+  );
+
+  App::RecordStream::Test::OperationHelper->do_match(
+    'parsedate',
+    [@args, qw[ --tz UTC --gmtime ]],
+    '{"when":"Feb 23 21:51:47 2016"}',
+    '{"when":"21:51:47"}',
+    "Feb 23 21:51:47 2016 (with --tz UTC) is 21:51:47 UTC",
+  );
+}
+
+note 'MDY vs DMY';
```

Comment:
Yep!  `note()` is only outputs when in verbose mode, so during a normal `make test` it won't show up.  Verbose mode provides detailed test output for each test, so it's useful to group them together under a heading (and `note()` ensures that its output doesn't interfere with the TAP stream).  Verbose mode is on when `TEST_VERBOSE` is true, usually because `-v` was passed to `prove` but we also set it in our Travis config.

Another useful testing function is `diag()`, which always outputs when called (again guaranteeing it won't interfere with the TAP stream, so it's nicer than `print` or `warn` that way).  It's most useful as a way to add more diagnostics to failures, usually with a pattern like:

```perl
is $got, $expected, "we got what we expected"
  or diag "Oh no, here's additional info about surrounding state: $info";
```

I think I've used that a couple places in recs so far.